sktime/forecasting/compose/_networkpipeline.py

```diff
@@ -0,0 +1,423 @@
+# -*- coding: utf-8 -*-
+# from sktime.forecasting.base._sktime import _SktimeForecaster
+# from sktime.forecasting.base._sktime import _OptionalForecastingHorizonMixin
+# from sktime.base import _HeterogenousMetaEstimator
+from sktime.forecasting.base._base import DEFAULT_ALPHA
+from sklearn.base import clone
+from sktime.forecasting.base import BaseForecaster
+
+__author__ = ["Viktor Kazakov"]
+__all__ = ["NetworkPipelineForecaster"]
+
+
+class NetworkPipelineForecaster(BaseForecaster):
+    """
+    Prototype of non sequential pipeline mimicking a network.
+
+    Each step in the pipeline can use the output of any of the previous steps or
+    the original input values to the fit and predict methods of the pipeline.
+
+    The arguments of the steps in the pipeline must be provided as a dictionary.
+    The dictionary is substituted in the fit_transform, fit or predict function calls
+    of the step transformer or estimator.
+
+    Parameters
+    ----------
+    steps : array of lists
+        list comprised of three elements:
+            1. name of step (string),
+            2. estimator (object),
+            3. arguments (dictionary) key value paris for
+            fit_transform or predict method of estimator
+
+    Examples
+    --------
+    `original_X` and `original_y` are key words referring to the
+    X and y arguments of NetworkPipelineForecaster.fit
+    and NetworkPipelineForecaster.predict
+
+    The arguments dictionary can refer to any of the previous
+    steps in the pipline by its name.
+
+    Forecasting example. Note convention for specifying
+    different arguments depending on whether
+    NetworkPipelineForecaster.fit or
+    NetworkPipelineForecaster.predict is called.
+    >>> from sktime.transformations.panel.dataset_manipulation import Selector
+    >>> from sktime.transformations.panel.dataset_manipulation import Converter
+    >>> from sktime.transformations.panel.dataset_manipulation import Concatenator
+    >>> from sktime.transformations.series.boxcox import BoxCoxTransformer
+    >>> from sktime.transformations.series.adapt import TabularToSeriesAdaptor
+    >>> from sklearn.preprocessing import MinMaxScaler
```

Comment:
```suggestion
```

Reply:
I am using MinMaxScaler in the example below:

```
    ... ("ft2", TabularToSeriesAdaptor(MinMaxScaler()), { "Z": "feature_X2"}),
```
and also

```
    ... ("new_y", TabularToSeriesAdaptor(MinMaxScaler()), {"Z":"original_y"}),
```